src/app/examples/crud/components/crud.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, NgForm } from '@angular/forms';
+
+import { takeUntil, map } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+
+import { ROUTE_ANIMATIONS_ELEMENTS } from '@app/core';
+
+import { State } from '../../examples.state';
+import { Book } from '../books.model';
+import {
+  ActionBooksPersist,
+  AddOne,
+  DeleteOne,
+  UpdateOne,
+  ActionBookSelect
+} from '../books.actions';
+import {
+  selectBooks,
+  selectCurrentBook,
+  selectAllBooks
+} from '../books.selectors';
+
+@Component({
+  selector: 'anms-crud',
+  templateUrl: './crud.component.html',
+  styleUrls: ['./crud.component.scss']
+})
+export class CrudComponent implements OnInit {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+
+  routeAnimationsElements = ROUTE_ANIMATIONS_ELEMENTS;
+
+  emptyBook: Book = {
+    id: new Date().getUTCMilliseconds().toString(),
+    title: '',
+    author: '',
+    description: ''
+  };
+  bookFormGroup = this.fb.group(this.emptyBook);
+  books: Book[];
+  selectedBook: Book | null;
+  isEditing: boolean;
+
+  constructor(public store: Store<State>, public fb: FormBuilder) {}
+
+  ngOnInit() {
+    this.store
+      .pipe(select(selectBooks), takeUntil(this.unsubscribe$))
+      .subscribe(books => {
+        this.store.dispatch(new ActionBooksPersist({ books }));
+      });
+    this.store
+      .pipe(
+        select(selectAllBooks),
+        takeUntil(this.unsubscribe$),
+        map(b => b.sort((b1, b2) => (b1.title > b2.title ? 1 : -1)))
```

Comment:
Would it make sense to this when we create the adapter? (it has a sort property)